### ISSUE TITLE
Add systemd support for the Ghidra server

### ISSUE THREAD

**Issue by fridtjof** [NationalSecurityAgency/ghidra#430]
**Is your feature request related to a problem? Please describe.**
I am trying to run the Ghidra server on Arch Linux, which only has systemd. Therefore there are no /etc/rcX.d and /etc/init.d directories, which causes yajsw to fail when starting the server.

**Describe the solution you'd like**
Because adding systemd support to yajsw is probably a bit out of scope for this, I suggest adding a script that directly launches the Ghidra server into foreground, so a systemd unit file can be written.

**Describe alternatives you've considered**
- Add systemd support to yajsw
- add a systemd unit file and more code to work around yajsw's behaviour

**Additional context**
With `set -x` enabled:
```
$ sudo ./ghidraSvr start
+ OPTION=start
+ '[' start == '' ']'
+ '[' 0 '!=' 0 ']'
+ APP_NAME=ghidraSvr
+ APP_LONG_NAME='Ghidra Server'
+ MODULE_DIR=Ghidra/Features/GhidraServer
+ WRAPPER_NAME_PREFIX=yajsw
+ SERVICE_NAME=org.rzo.yajsw.ghidraSvr
++ readlink -f ./ghidraSvr
+ SCRIPT_FILE=/opt/ghidra/server/ghidraSvr
+ SCRIPT_DIR=/opt/ghidra/server
+ cd /opt/ghidra/server
++ pwd
+ SCRIPT_DIR=/opt/ghidra/server
++ uname -s
+ OS=Linux
+ '[' Linux = Darwin ']'
+ OS_DIRNAME=linux64
+ '[' -d /opt/ghidra/server/../Ghidra ']'
+ GHIDRA_HOME=/opt/ghidra/server/..
+ WRAPPER_CONF=/opt/ghidra/server/server.conf
+ DATA_DIR=/opt/ghidra/server/../Ghidra/Features/GhidraServer/data
+ OS_DIR=/opt/ghidra/server/../Ghidra/Features/GhidraServer/os/linux64
+ CLASSPATH_FRAG=/opt/ghidra/server/../Ghidra/Features/GhidraServer/data/classpath.frag
+ LS_CPATH=/opt/ghidra/server/../support/LaunchSupport.jar
++ find /opt/ghidra/server/../Ghidra/Features/GhidraServer/data -maxdepth 1 -name 'yajsw*' -type d
++ head -n 1
+ WRAPPER_HOME=/opt/ghidra/server/../Ghidra/Features/GhidraServer/data/yajsw-stable-12.12
+ '[' '!' -d /opt/ghidra/server/../Ghidra/Features/GhidraServer/data/yajsw-stable-12.12 -o '!' -f /opt/ghidra/server/../Ghidra/Features/GhidraServer/data/yajsw-stable-12.12/wrapper.jar ']'
++ basename /opt/ghidra/server/../Ghidra/Features/GhidraServer/data/yajsw-stable-12.12
+ echo 'Using service wrapper: yajsw-stable-12.12'
Using service wrapper: yajsw-stable-12.12
+ JAVA_CMD=java
+ '[' -n '' ']'
+ JAVA_CMD=java
+ which java
++ java -cp /opt/ghidra/server/../support/LaunchSupport.jar LaunchSupport /opt/ghidra/server/.. -java_home
+ JAVA_HOME=/usr/lib/jvm/java-11-openjdk
+ '[' '!' 0 -eq 0 ']'
+ JAVA_CMD=/usr/lib/jvm/java-11-openjdk/bin/java
+ case "$1" in
+ checkInstall
++ java=/usr/lib/jvm/java-11-openjdk/bin/java
++ ghidra_home=/opt/ghidra/server/..
++ classpath_frag=/opt/ghidra/server/../Ghidra/Features/GhidraServer/data/classpath.frag
++ os_dir=/opt/ghidra/server/../Ghidra/Features/GhidraServer/os/linux64
++ /usr/lib/jvm/java-11-openjdk/bin/java -jar /opt/ghidra/server/../Ghidra/Features/GhidraServer/data/yajsw-stable-12.12/wrapper.jar -q /opt/ghidra/server/server.conf
++ grep 'Installed   :'
++ sed -E 's/Installed   : //'
WARNING: An illegal reflective access operation has occurred
WARNING: Illegal reflective access by org.codehaus.groovy.reflection.CachedClass (file:/opt/ghidra/Ghidra/Features/GhidraServer/data/yajsw-stable-12.12/./lib/extended/groovy/groovy-all-2.4.11.jar) to method java.lang.Object.finalize()
WARNING: Please consider reporting this to the maintainers of org.codehaus.groovy.reflection.CachedClass
WARNING: Use --illegal-access=warn to enable warnings of further illegal reflective access operations
WARNING: All illegal access operations will be denied in a future release
Apr 14, 2019 11:14:41 PM org.apache.commons.vfs2.VfsLog info
INFO: Using "/tmp/vfs_cache" as temporary files store.
Apr 14, 2019 11:14:41 PM org.rzo.yajsw.os.posix.PosixService init
WARNING: Error /etc/init.d : is not a directory
java.lang.reflect.InvocationTargetException
	at java.base/jdk.internal.reflect.NativeMethodAccessorImpl.invoke0(Native Method)
	at java.base/jdk.internal.reflect.NativeMethodAccessorImpl.invoke(NativeMethodAccessorImpl.java:62)
	at java.base/jdk.internal.reflect.DelegatingMethodAccessorImpl.invoke(DelegatingMethodAccessorImpl.java:43)
	at java.base/java.lang.reflect.Method.invoke(Method.java:566)
	at org.rzo.yajsw.boot.WrapperExeBooter.main(WrapperExeBooter.java:49)
Caused by: java.lang.NullPointerException
	at java.base/java.io.File.<init>(File.java:276)
	at org.rzo.yajsw.os.posix.PosixService.state(PosixService.java:662)
	at org.rzo.yajsw.wrapper.WrappedService.state(WrappedService.java:618)
	at org.rzo.yajsw.WrapperExe.doState(WrapperExe.java:592)
	at org.rzo.yajsw.WrapperExe.executeCommand(WrapperExe.java:176)
	at org.rzo.yajsw.WrapperExe.main(WrapperExe.java:124)
	... 5 more
+ RESULT=
+ '[' '' = true ']'
+ echo 'ERROR: ghidraSvr service is not installed'
ERROR: ghidraSvr service is not installed
+ return 1
+ '[' 1 = 0 ']'
```

**Comment by dev747368** [comment]
have you tried "./ghidraSvr console"?

**Comment by fridtjof** [comment]
Thank you very much - I just did, and that seems to do what I want.

For completeness, here's a basic systemd unit file using this that i wrote:
```
[Unit]
Description=Ghidra collaboration server

[Service]
ExecStart=/opt/ghidra/server/ghidraSvr console

[Install]
WantedBy=multi-user.target
```

Actually, now that it's running, I can't connect to it. I'll open a new issue/find an existing one.

**Comment by brainstorm** [comment]
TL;DR using docker & [systemd](https://tuttlem.github.io/2018/02/03/create-a-systemd-daemon.html) for my own future reference unless this is properly added upstream in Ghidra docs or somewhere else:

```shell
$ pwd
/usr/local/ghidra
$ cat bin/ghidra_server.sh
#!/bin/sh

docker run --rm --name ghidra-server \
	   -e GHIDRA_USERS="ubuntu another_user" \
	   -v /usr/local/ghidra/ghidra_collab:/repos \
	   -p 13100-13102:13100-13102 \
	   bytehow/ghidra-server:beta

$ cat  /lib/systemd/system/ghidra-server.service
[Unit]
Description=Ghidra Server
After=network.target

[Service]
ExecStart=/usr/local/ghidra/bin/ghidra_server.sh
Restart=on-failure

[Install]
WantedBy=multi-user.target

$ sudo systemctl daemon-reload
$ systemctl start ghidra-server
$ systemctl status ghidra-server
● ghidra-server.service - Ghidra Server
     Loaded: loaded (/lib/systemd/system/ghidra-server.service; enabled; vendor preset: enabled)
     Active: active (running) since Fri 2021-02-05 03:30:45 UTC; 3min 47s ago
   Main PID: 2162 (ghidra_server.s)
      Tasks: 12 (limit: 1111)
     Memory: 26.1M
     CGroup: /system.slice/ghidra-server.service
             ├─2162 /bin/sh /home/ubuntu/ghidra_server.sh
             └─2164 docker run --rm --name ghidra-server -e GHIDRA_USERS=ubuntu another_user -v /home>

Feb 05 03:31:01 ghidra-server ghidra_server.sh[2164]: 59/0|INFO     ubuntu (UserManager)
Feb 05 03:31:01 ghidra-server ghidra_server.sh[2164]: 59/0|INFO     another_user (UserManager)
Feb 05 03:31:01 ghidra-server ghidra_server.sh[2164]: 59/0|INFO  Known Repositories: (RepositoryMana>
Feb 05 03:31:01 ghidra-server ghidra_server.sh[2164]: 59/0|INFO     foo (RepositoryManager)
Feb 05 03:31:01 ghidra-server ghidra_server.sh[2164]: 59/0|INFO  Loading foo ... (Repository)
Feb 05 03:31:02 ghidra-server ghidra_server.sh[2164]: 59/0|INFO     ... loading 2 files ... (Reposit>
Feb 05 03:31:02 ghidra-server ghidra_server.sh[2164]: 59/0|INFO     foo load complete.  (Repositor>
Feb 05 03:31:02 ghidra-server ghidra_server.sh[2164]: 59/0|INFO  Starting Block Stream Server... (Bl>
Feb 05 03:31:02 ghidra-server ghidra_server.sh[2164]: 59/0|INFO  Registering Ghidra Server... (Ghidr>
Feb 05 03:31:02 ghidra-server ghidra_server.sh[2164]: 59/0|INFO  Registered Ghidra Server. (GhidraSe>
```

**Comment by cedws** [comment]
The fact that the `ghidraSvr console` forks still makes this an incredible pain in the ass. Legacy Java crap, yeugh.
